task/gitlab-jq/0.1/gitlab-jq.yaml

```
@@ -0,0 +1,99 @@
+        // if type == "array" then .[].id else .id end
+      description: >
+        The jq filter expression. See the jq manual for syntax: https://stedolan.github.io/jq/manual.
+        By default outputs the web_url of the returned object(s). Outputs the id if web_url is not available.
+    - name: DATA
+      default: ""
+      description: JSON body to send, if any.
+    - name: METHOD
+      default: ""
+      description: The HTTP method. Uses curl default if empty.
+    - name: JQ_EXTRA_ARGS
+      default: ""
+      description: Extra parameters to the jq command.
+    - name: GITLAB_TOKEN_SECRET_NAME
+      description: |
+        The name of the Kubernetes secret that contains the GitLab token
+      type: string
+      default: gitlab-api-secret
+    - name: GITLAB_TOKEN_SECRET_KEY
+      description: |
+        The key within the Kubernetes secret that contains the GitLab token
+      type: string
+      default: token
+    - name: GITLAB_HOST_URL
+      description: |
+        The GitLab host URL
+      default: "https://gitlab.com"
+      type: string
+    - name: API_PATH_PREFIX
+      description: |
+        The API path prefix
+      default: "api/v4"
+      type: string
+    - name: CURL_IMAGE
+      description: The image to send the request to the API.
+      default: docker.io/curlimages/curl@sha256:d6a01c11f0633375a173960ad741eca7460cbc776bb3bc2370e5571478b99459
+    - name: JQ_IMAGE
+      description: The image to filter the API response.
+      default: docker.io/stedolan/jq@sha256:a61ed0bca213081b64be94c5e1b402ea58bc549f457c2682a86704dd55231e09
+    - name: RESPONSE_BODY_PATH
+      description: Where to save the API response body before filtering it.
+      default: /workspace/gitlab_response.json
+  steps:
+    - name: request
+      image: $(params.CURL_IMAGE)
+      script: |
+        REQUEST_URL="$(params.GITLAB_HOST_URL)/$(params.API_PATH_PREFIX)/$(params.PATH)"
+        [ -n "$(params.METHOD)" ] && METHOD="-X $(params.METHOD)"
+        [ -n "$(params.DATA)" ] && DATA="--data '$(params.DATA)'"
+        curl -s -L --fail-with-body \
+          --header "User-Agent: TektonCD, the peaceful cat" \
+          --header "Authorization: Bearer $GITLAB_TOKEN" \
+          --header "Content-Type: application/json" \
+          $METHOD $DATA "$REQUEST_URL" \
+          -o $(params.RESPONSE_BODY_PATH) \
+        || {
+          cat $(params.RESPONSE_BODY_PATH)
+          exit 1
+        }
+      env:
+        - name: GITLAB_TOKEN
+          valueFrom:
+            secretKeyRef:
+              name: $(params.GITLAB_TOKEN_SECRET_NAME)
+              key: $(params.GITLAB_TOKEN_SECRET_KEY)
+    - name: filter
+      image: $(params.JQ_IMAGE)
+      script: |
+        #!/bin/bash
+        set -e
+        set -o pipefail
+        jq -r '$(params.FILTER)' $(params.RESPONSE_BODY_PATH) | tee $(results.output.path)
+  results:
+    - name: output
+      description: Output of the jq command.
```

should we change the description of result to something like `output of the gitlab api request filter` or something else?

---

```suggestion
      description: Output of the Gitlab API response filter.
```
I'd say "response" instead of "request"

---

makes sense